modules/document_repository/jsx/uploadForm.js

```diff
@@ -0,0 +1,197 @@
+import PropTypes from 'prop-types';
+import Loader from 'Loader';
+/**
+ * Media Upload Form
+ *
+ * Fetches data from Loris backend and displays a form allowing
+ * to upload a document file
+ *
+ * @author Shen Wang
+ * @version 1.0.0
+ *
+ * */
+class DocUploadForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      Data: {},
+      formData: {},
+      uploadResult: null,
+      errorMessage: null,
+      isLoaded: false,
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+    this.uploadFile = this.uploadFile.bind(this);
+    this.fetchData = this.fetchData.bind(this);
+  }
+ fetchData() {
+    return fetch(this.props.DataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({Data: data, isLoaded: true}))
+      .catch((error) => {
+        this.setState({error: 'An error occurred when loading the form!'});
+        console.error(error);
+      });
+}
+  componentDidMount() {
+    this.fetchData();
+  }
+  componentWillReceiveProps(nextProps) {
+    // Any time props.category changes, update state.
+    if (nextProps.category) {
+        this.fetchData();
+      }
+   }
```

Comment:
`componentDidMount` and `componentWillReceiveProps` should be listed before `fetchData`